src/metabase/query_processor/middleware/constraints.clj

```diff
@@ -5,26 +5,27 @@
    [metabase.models.setting :as setting]
    [metabase.util.i18n :refer [deferred-tru]]))
 
-(def ^:private ^:const default-max-results-bare-rows 2000)
-
-;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-results-bare-rows]])
-;; to a setting in 0.43 the setting, which allows for DB local value, can still be nil, so any places below that used
-;; to reference the former constant value have to expect it could return nil instead
 (setting/defsetting max-results-bare-rows
   (deferred-tru "Maximum number of rows to return specifically on :rows type queries via the API.")
-  :visibility     :authenticated
+  :visibility     :internal
   :type           :integer
-  :database-local :allowed)
+  :default        2000
+  :database-local :allowed
+  :getter         (fn [] (setting/get-value-of-type :integer :max-results-bare-rows)))
```

Comment:
You shouldn't need to add the `:getter`s, that's what the default is for an `:integer` setting anyway. This should work without them